widget/entry.go

```diff
@@ -461,12 +477,66 @@ func (e *Entry) Tapped(ev *fyne.PointEvent) {
 	e.Lock()
 	e.CursorRow = row
 	e.CursorColumn = col
+	if startSelect {
+		e.selectRow = row
+		e.selectColumn = col
+	}
 	e.Unlock()
 	Renderer(e).(*entryRenderer).moveCursor()
 }
 
-// TappedSecondary is called when right or alternative tap is invoked - this is currently ignored.
-func (e *Entry) TappedSecondary(_ *fyne.PointEvent) {
+// DoubleTapped is called when this entry has been double tapped so we should select text below the pointer
```

Comment:
This function has a lot of repeated code and complexity - I think some utility methods could be factored out.